infrastructure/template.yaml

```diff
@@ -173,22 +174,26 @@ Resources:
       LoggingConfig:
         LogGroup: !Sub /aws/lambda/${AWS::StackName}/CanaryInvokerFunction
       CodeSigningConfigArn:
-        !If [ EnforceCodeSigning, !Ref CodeSigningConfigArn, !Ref AWS::NoValue ]
+        !If [EnforceCodeSigning, !Ref CodeSigningConfigArn, !Ref AWS::NoValue]
       Policies:
         - Statement:
             Effect: Allow
             Action:
               - synthetics:StartCanary
               - synthetics:GetCanaryRuns
-            Resource: !Sub arn:aws:synthetics:${AWS::Region}:${AWS::AccountId}:canary:*
+            Resource:
+              - !Sub arn:aws:synthetics:${AWS::Region}:${AWS::AccountId}:canary:${NinoCICanary}
+              - !Sub arn:aws:synthetics:${AWS::Region}:${AWS::AccountId}:canary:${NinoHappyCanary}
+              - !Sub arn:aws:synthetics:${AWS::Region}:${AWS::AccountId}:canary:${Nino3rdPartyCICanary}
+              - !Sub arn:aws:synthetics:${AWS::Region}:${AWS::AccountId}:canary:${Nino3rdPartyHappyCanary}
 
   CanaryInvokerFunctionLogGroup:
     Type: AWS::Logs::LogGroup
     Properties:
       LogGroupName: !Sub /aws/lambda/${AWS::StackName}/CanaryInvokerFunction
-      RetentionInDays: !If [ IsProdEnvironment, 30, 7 ]
+      RetentionInDays: !If [IsProdEnvironment, "30", "7"]
```

Comment:
should we just set this to 30 days, I recollect some PR's changing all this to 30days in our templates. I don't know why?

Reply:
This was set like this by @surajkumar previously, I can make it 30 for all envs - I don't know if we have a specific policy

Reply:
Yeah, make 30 for all envs. Everything looks fine.